File: src/test/java/com/wxl/dyttcrawler/web/service/DyttStatisticServiceTest.java
package com.wxl.dyttcrawler.web.service;

import com.wxl.dyttcrawler.web.dto.TermItem;
import com.wxl.dyttcrawler.web.dto.statistic.BaseStatCount;
import com.wxl.dyttcrawler.web.dto.statistic.YearMonthCount;
import com.wxl.dyttcrawler.web.dto.statistic.YearPlaceCount;
import org.assertj.core.util.Lists;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import java.io.IOException;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

/**
 * Create by wuxingle on 2020/7/12
 */
@RunWith(SpringRunner.class)
@SpringBootTest
public class DyttStatisticServiceTest {

    @Autowired
    private DyttStatisticService statisticService;

    @Test
    public void testAggByField() throws IOException {
        TermItem<String, Long> termItem = statisticService.aggByField("category");
        System.out.println(termItem);
    }

    @Test
    public void testGetBaseStat() throws IOException {
        BaseStatCount baseStatCount = statisticService.getBaseStat();
        System.out.println(baseStatCount);
    }

    @Test
    public void testGetMonthCountGroupByYear() throws IOException {
        List<YearMonthCount> yearMonthCount = statisticService.getMonthCountGroupByYear(
                Lists.newArrayList(2018,2020));
        System.out.println(yearMonthCount);
    }

    @Test
    public void testGetPlaceCountGroupByYear() throws IOException {
        List<YearPlaceCount> yearPlaceCounts = statisticService.getPlaceCountGroupByYear(
                Collections.singletonList(2020));
        System.out.println(yearPlaceCounts);
    }
}

File: src/test/kotlin/com/wxl/dyttcrawler/web/service/DyttStatisticServiceTest.kt
package com.wxl.dyttcrawler.web.service

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * Create by wuxingle on 2020/7/12
 */
@ExtendWith(SpringExtension::class)
@SpringBootTest
class DyttStatisticServiceTest {

    @Autowired
    lateinit var statisticService: DyttStatisticService

    @Test
    fun testAggByField() {
        val termItem = statisticService.aggByField("category")
        println(termItem)
    }

    @Test
    fun testGetBaseStat() {
        val baseStatCount = statisticService.getBaseStat()
        println(baseStatCount)
    }

    @Test
    fun testGetMonthCountGroupByYear() {
        val yearMonthCount = statisticService.getMonthCountGroupByYear(
            listOf(2018, 2020)
        )
        println(yearMonthCount)
    }

    @Test
    fun testGetPlaceCountGroupByYear() {
        val yearPlaceCounts = statisticService.getPlaceCountGroupByYear(listOf(2020))
        println(yearPlaceCounts)
    }
}
